### PR TITLE
Add OADP 1.4.1 release notes

### PR DESCRIPTION
**Pull request by apurvabhide17**
Version(s):
OADP 1.4.1
OCP 4.14 → OCP 4.16

Issue:
[OADP-4740](https://issues.redhat.com/browse/OADP-4740)

Link to docs preview: 
https://81739--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/release-notes/oadp-1-4-release-notes.html#oadp-1-4-1-release-notes_oadp-1-4-release-notes

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:19]:
```suggestion
You can now change Velero Server Kubernetes API queries per second and burst values using the new Data Protection Application (DPA) fields. The new DPA fields are `spec.configuration.velero.client-qps` and `spec.configuration.velero.client-burst`, which both default to 100.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:28]:
```suggestion
Previously, the `PodVolumeBackup` custom resource (CR) had an incorrect message, `get a podvolumebackup with status "InProgress" during the server starting, mark it as "Failed"`.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:39]:
```suggestion
Previously, {oadp-short} set the `imagePullPolicy` parameter to `Always` for all images.
```
I think "previously ... used to... " is a bit repetitive

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:40]:
```suggestion
With this update, {oadp-short} checks if each image contains `sha256` or `sha512` digest, then it sets `imagePullPolicy` to `IfNotPresent`; otherwise sets it to `Always`. You can now override this policy by using the new `spec.containerImagePullPolicy` DPA field.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (rayfordj) [inline modules/oadp-1-4-1-release-notes.adoc:68]:
```suggestion
Previously, Velero memory leak was caused while using the CSI and DataMover plugins. When the backup ended, the Velero plugin instance was not deleted and it caused the memory to grow until OOM (Velero pod). With this update, there is no Velero memory leak.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:51]:
```suggestion
Previously, on performing a restore of the `BuildConfig` Build resource from a different cluster, the application resulted in errors on TLS verification to the internal image registry. It resulted in the `failed to verify certificate: x509: certificate signed by unknown authority` error.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (anarnold97) [inline modules/oadp-1-4-1-release-notes.adoc:68]:
```suggestion
Previously, Velero memory leak was caused while using the CSI and DataMover plugins. When the backup ended, the Velero plugin instance was not deleted and it caused the memory to grow till out-of-memory (OOM) (Velero pod). With this update, there is no Velero memory leak.
```

**Review by shdeshpa07 — COMMENTED** [review]
Please link to: https://docs.openshift.com/container-platform/4.14/openshift_images/triggering-updates-on-imagestream-changes.html

File: modules/oadp-1-4-1-release-notes.adoc
Comment (apurvabhide17) [reply modules/oadp-1-4-1-release-notes.adoc]: based on the existing occurrence in OCP, I've used Out of Memory (OOM).

File: modules/oadp-1-4-1-release-notes.adoc
Comment (stillalearner) [inline modules/oadp-1-4-1-release-notes.adoc:28]:
```suggestion
Previously, the `PodVolumeBackup` custom resource (CR) had an incorrect message, `: get a podvolumebackup with status "InProgress" during the server starting, mark it as "Failed"`.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (stillalearner) [inline modules/oadp-1-4-1-release-notes.adoc:45]:
```suggestion
With this update, the restore CR status would not be `InProgress` indefinitely and should eventually be `Completed` or `Failed`.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:61]:
```suggestion
With this update, the downloading of data proceeds to correct conclusion when restoring an empty PVC and the error message is not generated. 
```

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:61]: Suggested clarification.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:66]:
```suggestion
Previously, a Velero memory leak was caused by using the CSI and DataMover plugins. When the backup ended, the Velero plugin instance was not deleted and the memory leak consumed memory until an `Out of Memory` (OOM) condition was generated in the Velero pod. With this update, there is no resulting Velero memory leak when using the CSI and DataMover plugins.
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:77]:
```suggestion
While installing the OADP Operator in a namespace with more than 37 characters to create a new DPA, labeling the "cloud-credentials" Secret will fail and the DPA reports the following error:
```

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:77]: Suggesting simplifications because this is a complicated sentence. Emphasizing that the "37 characters" creates the problem and not creating a new DPA.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:84]:
```suggestion
.Restore is successfully completed by overriding the timeout error
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:86]:
```suggestion
Previously, in a large scale environment, the restore operation would result in a `Partiallyfailed` status with the error: `fail to patch dynamic PV, err: context deadline exceeded`.
```

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:86]: Adding "the" for clarification and the words "result in a" to provide clarification.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:77]: https://www.ibm.com/docs/en/ibm-style?topic=medium-global-audiences

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:98]:
```suggestion
After {oadp-short} restores, the Cassandra application pods may enter `CrashLoopBackoff` status. To work around this problem, delete the `StatefulSet` pods that are returning the error `CrashLoopBackoff` state after restoring {oadp-short}. The `StatefulSet` controller then recreates these pods and it runs normally.
```

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:98]: Suggestion for clarifying the logic.

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:98]: https://www.ibm.com/docs/en/ibm-style?topic=medium-global-audiences

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [inline modules/oadp-1-4-1-release-notes.adoc:105]:
```suggestion
During the restore operation, the {ocp} controller updates the `spec.template.spec.containers[0].image` field in the `Deployment` resource with an updated `ImageStreamTag` hash. The update triggers the rollout of a new pod, terminating the pod on which `velero` runs the FSB along with the post-hook. For more information about image stream trigger, see xref:../../../openshift_images/triggering-updates-on-imagestream-changes.adoc#triggering-updates-on-imagestream-changes[Triggering updates on image stream changes].
```

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:105]: Suggested simplification.

Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:105]: Suggesting post-hook instead of post restore hook.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (GroceryBoyJr) [reply modules/oadp-1-4-1-release-notes.adoc:69]: Suggesting Post-hook operation instead of Post restore hooks.

**Review by GroceryBoyJr — COMMENTED** [review]
@apurvabhide17 I have left a lot of comments, but they are very tiny nits about simplifying grammar and the word "post-hook". You have done excellent work on a very large and complicated release note. Please keep up the great work!

File: modules/oadp-1-4-1-release-notes.adoc
Comment (apurvabhide17) [reply modules/oadp-1-4-1-release-notes.adoc]: yup, thanks for sharing this.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (apurvabhide17) [reply modules/oadp-1-4-1-release-notes.adoc]: yup, I've removed it

**Comment by apurvabhide17** [comment]
> @apurvabhide17 I have left a lot of comments, but they are very tiny nits about simplifying grammar and the word "post-hook". You have done excellent work on a very large and complicated release note. Please keep up the great work!

Thanks @GroceryBoyJr for reviewing!

**Comment by apurvabhide17** [comment]
Hello @weshayutin, @stillalearner,
Requesting you to review the newly added two release notes in this [commit](https://github.com/openshift/openshift-docs/pull/81739/commits/b3eabf5ad9941ae30e49761ca99a75e24d238338). 

Once, I get an ack from bothof you, I'll share it again with @GroceryBoyJr for the ocp review.

Thanks!

File: modules/oadp-1-4-1-release-notes.adoc
Comment (stillalearner) [inline modules/oadp-1-4-1-release-notes.adoc:31]:
```suggestion
.Restoring a backup without pods is now successful
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (weshayutin) [inline modules/oadp-1-4-1-release-notes.adoc:6]: @anarnold97 and I thought it would be useful to add a link to FAQ in the release notes https://access.redhat.com/articles/5456281
probably at the top

Comment (apurvabhide17) [reply modules/oadp-1-4-1-release-notes.adoc:6]: ack, the link is now added. 
@weshayutin, are there any other suggestions for the added release notes or does it look good?

**Review by weshayutin — APPROVED** [review]
went through the rest of it and it looks good to me. Thank you very much!
/LGTM

**Comment by apurvabhide17** [comment]
> went through the rest of it and it looks good to me. Thank you very much! /LGTM

Thanks, @weshayutin!

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by apurvabhide17** [comment]
> LGTM
Thanks, @GroceryBoyJr!

File: backup_and_restore/application_backup_and_restore/release-notes/oadp-1-4-release-notes.adoc
Comment (anarnold97) [inline backup_and_restore/application_backup_and_restore/release-notes/oadp-1-4-release-notes.adoc:15]:
```suggestion
====
```

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Comment by GroceryBoyJr** [comment]
Hey @apurvabhide17 squash your commits and you are good to go to merge review!

**Comment by apurvabhide17** [comment]
> Hey @apurvabhide17 squash your commits and you are good to go to merge review!
@GroceryBoyJr, yup, I've squashed my commits after fixing the syntax issue and removing empty spaces, which you had suggested. 
Thank you for all your suggestions!

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:51]:
```suggestion
With this update, {oadp-short} checks if each image contains `sha256` or `sha512` digest, then it sets `imagePullPolicy` to `IfNotPresent`; otherwise `imagePullPolicy` is set to `Always`. You can now override this policy by using the new `spec.containerImagePullPolicy` DPA field.
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:51]: You need a noun here since you're using a semicolon. There must be two complete sentences on either side of a semicolon

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:57]:
```suggestion
With this update, the restore CR status for a restore correctly proceeds to the `Completed` or `Failed` status.
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:57]: IGMSG advises to avoid future tense

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:42]:
```suggestion
[source,text]
----
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:42]: You need this metadata per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#system-messages

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:69]:
```suggestion
[source,text]
----
```

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:83]:
```suggestion
Previously, due to the asynchronous nature of the Data Mover operation, a post-hook might be attempted before the Data Mover persistent volume claim (PVC) releases the persistent volumes (PVs) of the related pods. This problem would cause the backup to fail with a `PartiallyFailed` status.
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:83]: A suggestion to avoid possession for inanimate objects, per https://www.ibm.com/docs/en/ibm-style?topic=grammar-possessives

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:89]:
```suggestion
When you install the OADP Operator in a namespace with more than 37 characters to create a new DPA, labeling the "cloud-credentials" Secret fails and the DPA reports the following error:
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:89]: This sentence read a little strange to me. I wonder if "when you install" works as an alternative here? Also, avoid future tense as noted before.

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:93]:
```suggestion
With this update, creating a DPA does not fail in namespaces with more than 37 characters in the name.
```

Comment (skrthomas) [reply modules/oadp-1-4-1-release-notes.adoc:93]: use consistent language. You use create before, so use it again here instead of deploy

File: modules/oadp-1-4-1-release-notes.adoc
Comment (skrthomas) [inline modules/oadp-1-4-1-release-notes.adoc:110]:
```suggestion
After {oadp-short} restores, the Cassandra application pods might enter `CrashLoopBackoff` status. To work around this problem, delete the `StatefulSet` pods that are returning the error `CrashLoopBackoff` state after restoring {oadp-short}. The `StatefulSet` controller then recreates these pods and it runs normally.
```

**Review by skrthomas — COMMENTED** [review]
A few comments for you. Thanks! Nice work :)

**Review by stillalearner — APPROVED** [review]
lgtm

**Comment by bergerhoffer** [comment]
@apurvabhide17 Unfortunately it looks like this PR is in conflict, so you'll need to resolve that before we can merge. I'll continue with the rest of the merge review though, so that it can be good to go once you do.

Also, you have the PR marked as 4.13-4.16. But if it's targeting main, it'll need to go into 4.17 too. Just making sure that is okay with you.

**Comment by weshayutin** [comment]
The release notes for OADP-1.4.1 must go into the OCP 4.16 branch of the docs.  Thanks!

**Comment by bergerhoffer** [comment]
> The release notes for OADP-1.4.1 must go into the OCP 4.16 branch of the docs. Thanks!

Hi @weshayutin, my question is whether it is okay for this also to go into the 4.17 branch. (If the PR is against main, it also has to go into 4.17. It can't skip 4.17 and just do 4.16, 4.15, etc.).

Even if it might not apply to 4.17, we could always merge it as is and CP to 4.17, but then remove/adjust in 4.17 as needed.

**Comment by weshayutin** [comment]
Thanks for the clarification, I'm fine w/ main -> 4.17 -> 4.16  -> 4.15 etc..  
Thank you!

**Comment by bergerhoffer** [comment]
> Thanks for the clarification, I'm fine w/ main -> 4.17 -> 4.16 -> 4.15 etc.. Thank you!

Great thanks!

**Review by bergerhoffer — COMMENTED** [review]
This LGTM for merge, after the conflict is resolved!

I'm going to remove the merge-review-* labels for now. @apurvabhide17 once you resolve the conflict, you can either ping me directly for merge, or put it back in the merge queue. Thanks!

**Comment by apurvabhide17** [comment]
> @apurvabhide17 Unfortunately it looks like this PR is in conflict, so you'll need to resolve that before we can merge. I'll continue with the rest of the merge review though, so that it can be good to go once you do.
> 
> Also, you have the PR marked as 4.13-4.16. But if it's targeting main, it'll need to go into 4.17 too. Just making sure that is okay with you.

Hello @bergerhoffer, 
I've resolved the conflicts and added the merge label again. Could you please let me know if there are any other issues? 
Thanks

**Comment by bergerhoffer** [comment]
@apurvabhide17 4.13-4.15 failed to CP cleanly. Can you please do manual cherry picks for these versions?

**Comment by apurvabhide17** [comment]
> @apurvabhide17 4.13-4.15 failed to CP cleanly. Can you please do manual cherry picks for these versions?
Hello @bergerhoffer, 
Thanks for checking, could you please check the following PRs for this -

- For 4.13 - the OADP 1.4.0 release notes were added in 4.14-4.16 - [PR](https://github.com/openshift/openshift-docs/pull/78272), we can skip 4.13. I've updated the description of this PR.
- For 4.14, created this [PR](https://github.com/openshift/openshift-docs/pull/82190) for manual cherry-pick
- For 4.15, created this [PR](https://github.com/openshift/openshift-docs/pull/82191) for manual cherry-pick

**Pull request merged**
